File: ALGORITMOS/javascript/primo.js
//Questão 1
//Número é primo: Um número inteiro positivo n é primo se for divisível apenas por 1 e por n.

function primo(numeros) {
    const primos = [];
    for (let i = 0; i < numeros.length; i++) {
        let num = numeros[i];
        if (num < 2) { continue };
        let primo = true;
        for (let j = 2; j <= Math.sqrt(num); j++) {
            if (num % j === 0) {
                primo = false;
                break;
            }
        }
        if (primo) {
            primos.push(num);
        }
    }
    return primos;
}

module.exports = primo;

File: ALGORITMOS/java/Primo.java
//Questão 1
//Número é primo: Um número inteiro positivo n é primo se for divisível apenas por 1 e por n.

public class Primo {
    public void verificarPrimos(int[] numeros) {
        for (int num : numeros) {
            if (num < 2) {
                continue;
            }
            boolean primo = true;
            for (int i = 2; i <= Math.sqrt(num); i++) {
                if (num % i == 0) {
                    primo = false;
                    break;
                }
            }
            if (primo) {
                System.out.println(num + " é primo.");
            }
        }
    }
}
